### ISSUE TITLE
Search box bug

### ISSUE THREAD

**Issue by archqob** [LinxGem33/Arc-Menu#150]
When i press space it launch the selected application rather than type space and wait for next word to be written.

**For example:** when i try to type 'Text Editor' i write 'Text' then press space.

- **Expected behavior:** type space and wait for the next word to be written "keep focus on search box".
- **Real behavior:** launch selected application and close arc-menu.

**Comment by LinxGem33** [comment]
@archqob 

appreciate the feed back, i can confirm that i was able to replicate this behaviour we will investigate this issue and ill also issue a priority for this as i think this will affect the majority of users we will contact you should we need further information regarding this issue.

kind regards

**Comment by lexruee** [comment]
**@LinxGem33 @archqob** 
Isn't this the same issue as I already noted here https://github.com/LinxGem33/Arc-Menu/issues/40 ?

**Comment by LinxGem33** [comment]
@lexruee 

honestly it was so long ago regarding this issue #40 i actually forgot that we was going to try and implement the enter instead of the space bar as there has been so many changes since then however its now been classed as a bug as the space bar is not normal behaviour for its use.

**Comment by lexruee** [comment]
@LinxGem33 @archqob 
I created a `fix-search-box-bug` branch on my repository which fixes this issue:
https://github.com/lexruee/Arc-Menu/tree/fix-search-box-bug

But before I make a pull request, I would like to evaluate how this search box feature could be overall improved. I'm thinking of abstracting and improving the search algorithm (see [menu.js#L1146](https://github.com/LinxGem33/Arc-Menu/blob/Development-branch/menu.js#L1146)).
Currently, the search algorithm is still based on the Zorin menu's implementation.

In short, the algorithm creates for each application a text string containing multiple terms that characterize the application. The entered query or search value by the user is then checked against the text string by using a simple `indexOf` lookup.

In the end searching for applications is just an information retrieval problem :smile:. So we could easily apply some IR techniques here :-).

**Comment by LinxGem33** [comment]
@lexruee 
i think that is a great idea it would also allow us to examine the possibilities regarding some of the search features, i will monitor the `fix-search-box-bug`  branch hopefully we can implement some extra search functionality as suggested :+1:

**Comment by LinxGem33** [comment]
@archqob 

the fix has now been issued by @lexruee (please see [bug fix](https://github.com/lexruee/Arc-Menu/commit/1f2fb19ce680432ce12c46cb1c25b21af1a6a334)) and will be merged into the development branch once a few more refinements have been made, because this issue was a high priority issue we aim to fix these issues in a timely manner hence the quick bug fix.

kind regards
